src/components/FilterAnnotMondal/FilterAnnotModal.js

```diff
@@ -0,0 +1,172 @@
+      if (core.getDisplayAuthor(annot) && core.getDisplayAuthor(annot) !== '') {
+        authorsToBeAdded.add(core.getDisplayAuthor(annot));
+      }
+      if (annot.Subject && annot.Subject !== '') {
+        // we want to lowercase it and remove spaces to get i18n translations
+        annotTypesToBeAdded.add(annot.Subject.toLowerCase().replace(/\s/g, ''));
+      }
+    });
+    setAuthors([...authorsToBeAdded]);
+    setAnnotTypes([...annotTypesToBeAdded]);
+  }, [isOpen]);
+
+  const modalClass = classNames({
+    Modal: true,
+    FilterAnnotModal: true,
+    open: isOpen,
+    closed: !isOpen,
+  });
+
+  return isDisabled ? null : (
+    <Swipeable onSwipedUp={closeModal} onSwipedDown={closeModal} preventDefaultTouchmoveEvent>
+      <div className={modalClass} data-element="filterModal" onMouseDown={closeModal}>
+        <FocusTrap locked={isOpen} focusLastOnUnlock>
+          <div className="container" onMouseDown={e => e.stopPropagation()}>
+            {core.getAnnotationsList().length > 0 ? (
+              <div className="filter-modal">
+                <div className="swipe-indicator" />
+                <div className="filter-options">
+                  <div className="filter">
+                    <div className="heading">{t('option.filterAnnotModal.commentBy')}</div>
+                    <div className="buttons">
+                      {[...authors].map((val, index) => {
+                        return (
+                          <Choice
+                            type="checkbox"
+                            key={index}
+                            label={val}
+                            checked={authorFilter.includes(val)}
+                            id={val}
+                            onChange={e => {
+                              if (authorFilter.indexOf(e.target.getAttribute('id')) === -1) {
+                                setAuthorFilter([...authorFilter, e.target.getAttribute('id')]);
+                              } else {
+                                setAuthorFilter(authorFilter.filter(author => author !== e.target.getAttribute('id')));
+                              }
+                            }}
+                          />
+                        );
+                      })}
+                    </div>
+                  </div>
+                  <div className="filter">
+                    <div className="heading">{t('option.filterAnnotModal.types')}</div>
+                    <div className="buttons">
+                      {[...annotTypes].map((val, index) => {
+                        return (
+                          <Choice
+                            type="checkbox"
+                            key={index}
+                            label={
+                              t(`annotation.${val}`)
+                                .toLowerCase()
+                                .replace(/\s/g, '') === val
+                                ? t(`annotation.${val}`)
```

Comment:
See my another comment regarding if we can just display `annot.Subject` here instead of getting the `val`s and use `t(annotation.${val})`